cvcreator/vitae/tech_skills.py

```diff
@@ -92,3 +88,49 @@ def make_skill_groups(
     output = [TechnicalSkill(title=title, values=output[title])
               for title in skills_data["allowed_labels"] if title in output]
     return output
+
+
+def get_skills(skill_categories: List[SkillCategory]) -> List[TechnicalSkill]:
+    """Process skills organized in skill categories into a list of `TechnicalSkill` objects
+
+    Args:
+        skill_categories:
+            User provided skills to distribute into groups.
+
+    Returns:
+        List of TechnicalSkill objects
+    """
+    output: List[TechnicalSkill] = []
+
+    # Get technical skills data
+    skills_data = get_skills_data()
+
+    # Process the skill categories in the CV
+    for skcat in skill_categories:
+        label = skcat.category
+        skills = skcat.technical_skills
+        # Validation of skill categories and skills
+        assert label in skills_data["allowed_labels"], ValueError(f"{label}: Invalid skill label")
+        _validate_skills(skills, skills_data)
+
+        output.append(TechnicalSkill(title=label, values=skills))
+
+    return output
+
+
+def _validate_skills(skills: Sequence[str], skills_data: dict[str, Any]):
+    """Check that skills are valid.
+
+    Compare a list of skill strings against a data structure containing allowed skills.
+
+    Args:
+        skills:
+             List of skills.
+        skills_data:
+            Technical skills data.
+    """
+    skills = list(skills)
+    unknown_skills = set(skills).difference(skills_data["skills"])
```

Comment:
I see this comes from older code, but is listifying the `skills` variable necessary here? Since it is converted to a set in the next line anyway and is not being used later. What do you think?

```suggestion
    unknown_skills = set(skills).difference(skills_data["skills"])
```